### ISSUE TITLE
Refactor the content script to inject an object in the window instead

### ISSUE THREAD

**Issue by tomaka** [paritytech/substrate-connect#678]
After #572:

- [ ] Move the messages of the "connect-protocol" (the messages that #572 are about) to the `extension` project.
- [ ] Completely wipe "connect-protocol" and expose just one interface that contains an `addChain` function.
- [ ] Refactor the extension's content script to not listen for `window` messages anymore, but instead inject an object that implements this interface into `window`, so that the web page can use it.
- [ ] Refactor the `ExtensionProvider` appropriately.

I put several bullet points, but it doesn't makes sense to me to tackle them separately. This can be done with several commits in a single PR.

**Comment by tomaka** [comment]
Closing in favor of https://github.com/paritytech/substrate-connect/issues/712
